cmd/erasure-object.go

```diff
@@ -943,7 +943,7 @@ func (er erasureObjects) putObject(ctx context.Context, bucket string, object st
 
 	if opts.CheckPrecondFn != nil {
 		obj, err := er.getObjectInfo(ctx, bucket, object, opts)
-		if err != nil {
+		if err != nil && !isErrVersionNotFound(err) {
```

Comment:
shouldn't the caller be ignoring this? not here?

Reply:
ETag was not being passed into the precondition match correctly from opts in #15674, so it worked previously for the wrong reason - when a replica PUT event is received, the version will not exist and should be treated as ok